### ISSUE TITLE
Azure Cli Tasks - Independent of each other in a build pipeline?

### ISSUE THREAD

**Issue by rysantos** [Microsoft/azure-pipelines-tasks#8837]
## Environment
- Server - Azure Pipelines
- Agent - Hosted
- Azure Cli Version 1.143.2

## Issue Description
My team and I are currently delivering a project pipeline that used two separate azure-cli tasks. 

- One was to install the azure-cli extension botservice. 

- One was to run msbot clone. 

This worked until our pipeline was updated to version 1.143.2 in this commit: 
https://github.com/Microsoft/azure-pipelines-tasks/commit/f29fe658756877e9f999852d384af0b68def0f1b#diff-4321c25c64999e4bfddfae8a7311587f
As a result our second msbot clone task was failing because it wasn't recognizing the extension added in the previous task. 

I am not sure if this was the intended impact or if we configured our solution incorrectly. Ideally I would like to see some documentation on how to execute Azure Cli Tasks properly. (Maybe I am just missing it?) From my vague understanding of the source code, it looks like Azure Cli tasks are now running in their own directory ("azclitask") and are unrelated to each other? Not sure if my terminology is correct here... Now our second task cannot see the extension that the first task added. 

This was the set up of our build pipeline:

- Task 1 - Add Bot Service Extension
 ![image](https://user-images.githubusercontent.com/43550008/48467068-74383800-e7ad-11e8-96fe-6ea64ce67af3.png)

- Task 2 - msbot clone 
![image](https://user-images.githubusercontent.com/43550008/48467093-874b0800-e7ad-11e8-925a-d1f909b57db5.png)

### Task logs

- Logs for Task 1 - Add Bot Service Extension
As you can see from the following log, the bot service extension is added to the extensions directory at 'C:\Users\VssAdministrator\.azclitask\c1542181244092\cliextensions' as opposed to the global directory of 'C:\Users\VssAdministrator\.azure\cliextensions'. 

[3_Add botservice extension.txt](https://github.com/Microsoft/azure-pipelines-tasks/files/2579660/3_Add.botservice.extension.txt)

- Logs for Task 2 - msbot clone
msbot clone runs the command `az -v` to check versioning. This looks for the extension in the tasks' current directory and fails to find the updated botservice. (Because it was added in the other task 1's azclitask directory) 

[4_Create bot resources.txt](https://github.com/Microsoft/azure-pipelines-tasks/files/2579664/4_Create.bot.resources.txt)

### Solution
Our solution was simple: 
We just consolidated the two tasks into a single one: 
![image](https://user-images.githubusercontent.com/43550008/48467963-1f49f100-e7b0-11e8-9fc8-e11e03d39a71.png) 
However, I could see this becoming a problem if tasks needed to be separated for some reason. It also could be a problem for legacy pipelines running on previous azure-cli task builds. Ours just starting failing yesterday without any changes. I don't believe I can specify which version of 1.* to target in the UI. 

Long story short, wanted to make you all aware of this because this took a significant amount of time to debug and identify in our solution. Also wanted to help the rest of the community in case anyone else is experiencing a similar issue.

**Comment by sachinma** [comment]
Yes, your understanding is right. This change was made to address the scenario where two agents might be running on the same machine. Without the fix - logging out from one task, would cause other pipeline to fail. 

Thanks for sharing the solution to the community and apologies for the issue - we missed anticipating your use-case while building this feature.

I will keep this issue open and will consider the option to let the user select the behavior.

**Comment by bishal-pdMSFT** [comment]
Another option is to set [AZURE_EXTENSION_DIR](https://docs.microsoft.com/en-us/cli/azure/azure-cli-extensions-overview?view=azure-cli-latest#install-extensions) environment variable to a common directory e.g. $HOME/.azure/cliextensions.
Task can do this while setting a new config directory which will ensure that extensions are installed/loaded at a common directory.

**Comment by vineetmimrot** [comment]
This issue has been fixed https://github.com/Microsoft/azure-pipelines-tasks/pull/8998. It will be rolled out by tomorrow.

**Comment by vineetmimrot** [comment]
fix is available in 1.144.3 version.

**Comment by bfbatch** [comment]
Hi @rysantos, wondering if we can connect. I'm just starting a chatbot project and was looking for your best practices.
